src/data/error.rs

```diff
@@ -333,6 +330,9 @@ pub enum LexError {
     #[error("underflow parsing floating literal")]
     FloatUnderflow,
 
+    #[error("{0}")]
+    InvalidHexFloat(hexponent::ParseError),
```

Comment:
Can you use `#[from]` here? Or does ParseError not implement Error?

Reply:
I can't use from because it requires an extra method not on `std::error::Error` that is implemented by the proc-macro.

Reply:
That doesn't match the documentation ...

> Any error type that implements std::error::Error or dereferences to dyn std::error::Error will work as a source.

https://docs.rs/crate/thiserror/1.0.9